src/components/SubjectsPanel/index.tsx

```
@@ -20,31 +22,71 @@ export function SubjectsPanel({ onOpenSubject, onSearchSubject, onOpenStudyPlan
+  // Erasmus/exchange students have no KontrolaPlanu, so the plan never
+  // materializes — fall back to the subjects store (student/list.pl).
+  const planUsable = !!plan && plan.blocks.some(b => b.groups.some(g => g.subjects.length > 0));
+  // The fallback must wait until the plan has actually settled (loaded, handshake
+  // resolved, not mid-sync) — otherwise regular students see the exchange caption
+  // flash during cold load, since the subjects store (Tier 1) populates before the
+  // study plan (Tier 2 microtask). Latch: background resyncs flip isSyncing every
+  // few minutes; once settled once, later isSyncing:true phases must not collapse
+  // the fallback back to a skeleton mid-session for Erasmus students.
+  const planSettled = studyPlanLoaded && (handshakeDone || handshakeTimedOut) && !isSyncing;
+  const settledOnce = useRef(false);
+  if (planSettled) settledOnce.current = true;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/SubjectsPanel/index.tsx:42:20
  40 |   const planSettled = studyPlanLoaded && (handshakeDone || handshakeTimedOut) && !isSyncing;
  41 |   const settledOnce = useRef(false);
> 42 |   if (planSettled) settledOnce.current = true;
     |                    ^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  43 |   const gateOpen = planSettled || settledOnce.current;
  44 |   const fallbackPlan = useMemo(() => {
  45 |     if (!gateOpen || planUsable || !subjects || Object.keys(subjects.data).length === 0) return null;

(react-hooks/refs)